0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -56,3 +52,123 @@ All ILP Packets MUST be returned with the HTTP status code `200: OK`.
+
+A JWT token SHOULD also include an `exp` (expiry) claim that indicates a date/time after which the token should be considered invalid. Implementations SHOULD reject any tokens with a missing or invalid expiry claim. 
+
+Note that tokens without this claim never expire, and only become invalid if the shared-secret used to sign the JWT changes or is otherwise invalidated.
+
+##### Example Usage
+In this profile, the Base64-encoded JWT is passed as an `Authorization` header in each HTTP request, using the [Bearer token](https://tools.ietf.org/html/rfc6750) scheme. 
+
+One example of such a Bearer token:
+
+`Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhbGljZSJ9._Jn0pkqrK1leE3WZJKn-g5hm5kGJxGdSHggtz5wO1w4`. 
+
+Using the JWT specification, this token can be verified using the shared-secret previously agreed upon. For example, the above token contains a `sub` claim of `alice` and can be verified using a shared-secret value of `HEiMCp0FoAC903QHueY89gAWJHo` (Base64 encoded).
+
+Another example is a Bearer token that contains an expiration date:
+
+`Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhbGljZSIsImV4cCI6MTU1ODAzNTg2OH0.__9CiSGdn4Grhl48slun7Lp4q4xt0uq398omcipBU8M`. 
+
+Using the JWT specification, this token can be verified using the shared-secret previously agreed upon. For example, the above token contains a `sub` claim of `alice` and an `exp` claim of `1558035868`, which means this token is no longer valid after `May 16th, 2019 at 9:29:11 GMT`. This token can be verified using a shared-secret value of `HEiMCp0FoAC903QHueY89gAWJHo` (Base64 encoded).
+
+##### Trade-off Summary
+* **Pros**
+  * Same usability as the `SIMPLE` profile -- just a shared-secret with _at least_ 32 bytes and a `sub` claim.
+  * Allows both identity and authentication claims to be contained in single Bearer token, which eliminates the need for a second `Auth-Principal` header. This should simplify token validation and ease account-lookup (no need to create an HMAC or encrypted data-store index because the principal can be trivially decoded from the token).
+  * Requires only enough processing to perform an HMAC-SHA256 signing operation, which is very fast.
+  * Supports token expiry, which allows tokens to be generally short-lived so that peers can narrow the potential window of unauthorized usage in the event of token compromise.
+  * The actual shared-secret is _never_ transmitted "on the wire" during any request. Instead, authentication tokens are always _derived_ from the shared-secret, which eliminates the risk of an _actual_ shared-secret being intercepted  in transit.
+
+* **Cons**
+  * Total transmitted bytes for authentication are more than the `SIMPLE` scheme (about 41 bytes, or ~50% more). However, HTTP/2 header compression should mitigate this differential.
+  * Requires out-of-band communication for both peers to agree upon the shared secret.
+
+## Appendix1: Security Best Practices
```

Comment:
@adrianhopebailie Do you feel fine leaving this appendix in this RFC? I tried to keep the recommendations specific to tokens, so this is _probably_ the right place...but am open to other ideas.

Reply:
The goal is interop so we should define the minimum requirements of implementations to be interoperable knowing that that will become the most commonly used profile so it should be well designed.

Client certs are def too complex to make a normative requirement.

Reply:
> we should define the minimum requirements of implementations to be interoperable

Totally agree - I made the intro to this section (i.e., `Appendix1`) very overt to say that these are only recommendations, and are all non-normative. 

I guess my question was: is this RFC the right place to put a bunch of non-normative security recommendations, or does that kind of thing belong somewhere else?

Reply:
No, I think this is a fine place. 

Although in some respects I'd say we could use this same auth stuff for WebSocket based transports too so maybe a stand-alone spec will be the way we go in the long run...

Reply:
> No, I think this is a fine place.

Cool, I'll keep it for now.